packages/gitbook/src/components/DocumentView/CodeBlock/highlight.ts

```diff
@@ -330,7 +352,7 @@ const createHighlighter = createdBundledHighlighter<any, any>({
  * Load the highlighter, only once, and reuse it.
  * It makes sure to handle concurrent calls.
  */
-const loadHighlighter = singleton(async () => {
+const loadHighlighter = memoize(async () => {
```

Comment:
This part was a singleton and now it's memoized because I can't use server-side code in a client-component, even if we don't really use it.

Reply:
I solved it by using a server action.